### ISSUE TITLE
[Request] (Android) Add controller support for Shanwan Usb Controller

### ISSUE THREAD

**Issue by Leopard1907** [hrydgard/ppsspp#11529]
Controller:

https://forums.libretro.com/t/shanwan-pc-ps3-android-wireless-controller-on-lakka/8887/6

https://retropie.org.uk/forum/topic/16401/tgz-850m-shanwan-android-2-4ghz-wireless-controller-hotkey-config/7
A PS3 clone wireless controller ( via it's dongle) which also have rumble support.

Rumble really doesn't matter much though.

### What happens?

My controller doesn't work with PPSSPP via OTG on Android phone.

### What *should* happen?

If it is possible , a working controller on my end

### What hardware, operating system, and PPSSPP version?  On desktop, GPU matters for graphical issues.

LG G2 D802 
Android 5.0.2 ( Official rom if that matters)
Kernel 3.4.0
PPSSPP version : 1.7.2
Installed through Google Play Store

**Comment by Florin9doi** [comment]
Your link contains a kernel patch which cannot be applied by a user mode app.

**Comment by Leopard1907** [comment]
I gave these links to easen out identifying controller.

Not for " apply these patches " .

**Comment by hrydgard** [comment]
Been meaning to get a bunch of random Android controllers to sort out any remaining systemic issues, but I can't find this one to buy anywhere. Can you point me to some place that sells them?

**Comment by Leopard1907** [comment]
I searched it for aswell. I couldn't find it either.

I bought these controllers years ago , from a local shop. Mine were branded as Tasco Gamepad. Like any usual Chinese knockoff controller , it is rebranded under many names.

How can i know the model then?

1-) Steam labels it under that name in controller configs.
2-) RPCS3 also recognizes it with that Shanwan name. It uses Evdev for input , at least on Linux.

I have 4.15 kernel on my Linux machine.

**Comment by xiushudongfang** [comment]
I found a home of that 山弯（shanwan）company.
http://www.bmchip.com/

**Comment by hrydgard** [comment]
http://www.bmchip.com/te_Product_bt/2015-04-15/4210.chtml seems to be the controller, but no way to order, I guess they only deliver to distributors (and most likely have discontinued them)..

**Comment by Leopard1907** [comment]
It is not that product. 

Mine is a standart PS3 controller with no Android support claiming at all. 

My controller has an USB 2.4 ghz wireless receiver dongle which i connect to my phone via OTG.

That controller works for Epsxe on Android.

**Comment by hrydgard** [comment]
Oh, right, OTG...

**Comment by Leopard1907** [comment]
I do not have access to that phone anymore and i don't use that controller anymore to test on another device as well. So closing this issue as it only sits here without a purpose.

**Comment by AndrewPrzelucki** [comment]
Hi, is there any chance for a driver for a wireless PC/PS3 gamepad that is identified under Linux as 2563:0526 (vendor:product)? The pad is named "SHANWAN Android Gamepad" and from what I found out it's made by ShenZhen ShanWan Technology. The Windows driver is [here (link)] (https://esperanza.pl/esperanza-gamepad-bezprzewodowy-2-4ghz-ps3-pc-usb-gladiator-czarny,176,1697.html). The gamepad offers rumble but it's not working under Linux, and also the analog/digital switch isn't working under Linux. I found a driver on [GitHub (link)](https://github.com/hbiyik/hid-shanwan) but it's for a different product ID, namely 2563:0575. The ShenZhen ShanWan Technology Co., Ltd webpage lists a few they make and I found a similar one [here (link)](http://www.bmchip.com/te_product_mis/2012-08-20/2011.chtml), but I don't have any information about the product number other than that mine is 2563:0526 but the webpage calls it BM-977. Any information is welcomed, maybe someone knows about a Linux driver somewhere. I found a [python script (link)](https://github.com/paroj/xpad/issues/83) for another model, but it doesn't work for my product ID. Thanks!
